twap/price/price.go

```diff
@@ -0,0 +1,48 @@
+package price
+
+import (
+	"fmt"
+	"sort"
+	"time"
+)
+
+type Price struct {
+	Value     float64   `json:"price"`
+	Timestamp time.Time `json:"timestamp"`
+}
+
+func TWAP(samples []Price) (float64, error) {
```

Comment:
I created a TWAP utility function with its own testing. Initially I played with the idea that the TWAP would be calculated up to the current time (passed in as an additional parameter). I think that's the right way to do it, but it made the value of the TWAP change whenever I called this function, since it depended on the samples, but also the current time that would change the weight of the most recent sample.
To avoid the dynamic behavior, I changed the function to its current form, where we calculate the twap based on the samples alone. That makes the value of the least recent function not matter, as you see in the comment below.

Reply:
@dlm Could you pay particular attention to the implementation of this function in your review?